src/build.rs

```diff
@@ -171,6 +171,13 @@ impl BuildDirectory {
         Ok(())
     }
 
+    /// Get the path to the source code on the host machine (outside the sandbox).
+    pub fn get_source_dir(&self, krate: &Crate) -> Result<PathBuf, Error> {
```

Comment:
Hmm, I'd still just make the `Crate::copy_source_to` method public instead of an approach like th is.

Reply:
Where does one get the target dir from? `krate.copy_source_to(&workspace, ???)`

Reply:
Docs.rs would have to create a temporary dir (for example with the `tempfile` crate) to extract the source code to.